packages/context/src/context.ts

```diff
@@ -53,6 +53,12 @@ import {iterator, multiple} from 'p-event';
 
 const debug = debugFactory('loopback:context');
 
+export type ContextEventListener = (
+  binding: Readonly<Binding<unknown>>,
+  context: Context,
+  event: string,
+) => void;
```

Comment:
Can you please explain what's the difference between `ContextObserver` and `ContextEventListener`? When should our users use which?

Please add a tsdoc comment for this interface. I think we should update the documentation to mention this new way of observing context changes.

Reply:
`ContextEventListener` is a sync function as typical Node.js event listeners. The `ContextObserver` is async handler.

Reply:
> `ContextEventListener` is a sync function as typical Node.js event listeners. The `ContextObserver` is async handler.

Sure, I can tell this part from the source code. I am asking for user-centric documentation - when should our users choose ContextObserver and when to choose ContextEventListener? As framework authors, we should set clear guidance here.